src/webapi/node.rs

```diff
@@ -362,6 +362,45 @@ pub struct Node( Reference );
 impl IEventTarget for Node {}
 impl INode for Node {}
 
+impl Node {
+    /// Attempt to create the `Node` from raw html.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if the resulting number of nodes != 1. To ensure this
+    /// it is recommended to at _least_ have programatic control of your root
+    /// HTML node.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let node = Node::from_html("<div>Some text, horray!</div>").unwrap();
+    /// ```
+    pub fn from_html(html: &str) -> Result<Node, SyntaxError> {
+        let result: Result<Value, Value> = js_try!(
+            var span = document.createElement("span");
+            span.innerHTML = @{html};
+            if( span.childNodes.length != 1 ) {
+                throw SyntaxError("HTML had ${span.childNodes.length} nodes but must have 1");
```

Comment:
We should not use ES6 features in `js!` snippets. (:

Also, perhaps an error like this would be better? `"Node::from_html called on a snippet with multiple root nodes"`

Reply:
what's the ES6 feature?

Reply:
Also, you would get the error if you have 0 nodes as well, which is why I included the number.

Reply:
> what's the ES6 feature? (I'm such a js noob)

The string interpolation. `${span.childNodes.length}`

> Also, you would get the error if you have 0 nodes as well, which is why I included the number.

Hmm.. yes, you're right. Then perhaps we could have an inner `if` which would check for that? Or just say that `"Node::from_html requires a single root node"`? I don't really care *that* much either way though, as long as the error message is clear what went wrong.

Reply:
I did

                throw SyntaxError(
                    "Node::from_html requires a single root node but has: "
                    + toString(span.childNodes.length)
                );

Reply:
Still having issues with the test failing though. Don't know why (says that the resulting error is `CustomError` instead of `SyntaxError`, even when I'm throwing `SyntaxError`)

Reply:
I don't think `toString` is valid, you should just remove it:
```
throw SyntaxError(
    "Node::from_html requires a single root node but has: " + span.childNodes.length
);
```

JS automatically converts things into strings when you try to add it to a string.